src/main.rs

```diff
@@ -1,74 +1,132 @@
-use database::connection_pool;
-use deadpool_postgres::{Client, Pool};
-use error::ErrorReply;
-use rweb::{get, openapi, openapi_docs, rt::tokio, serve, Filter, Json, Rejection, Reply};
+#[macro_use]
+extern crate diesel;
+#[macro_use]
+extern crate diesel_migrations;
+
+use actix_web::{get, middleware::Logger, post, web, App, HttpResponse, HttpServer, Responder};
+use database::migrate_database;
 use uuid::Uuid;
 
 use crate::{
-    database::{get_graph_by_id, test_connection},
+    database::PgPool,
     error::Error,
-    scoring::{ScoreGraph, Scores},
+    models::{Dataset, Dimension},
+    score::SaveRequest,
 };
 
 mod database;
 mod error;
-mod scoring;
-mod vocab;
+mod models;
+mod schema;
+mod score;
 
 #[get("/ping")]
-async fn ping(#[data] pool: Pool) -> Result<impl Reply, Rejection> {
-    async fn ping(pool: Pool) -> Result<impl Reply, Error> {
-        let client: Client = pool.get().await?;
-        test_connection(&client).await?;
-        Ok("pong")
-    }
-
-    Ok(ping(pool).await?)
+async fn ping(pool: web::Data<PgPool>) -> Result<impl Responder, Error> {
+    let conn = pool.get()?;
+    conn.test_connection()?;
+    Ok("pong")
 }
 
 #[get("/ready")]
-fn ready() -> impl Reply {
-    "ok"
+async fn ready() -> Result<impl Responder, Error> {
+    Ok("ok")
+}
+
+#[get("/api/v1/graph/{id}")]
+async fn get_score_graph(
+    id: web::Path<String>,
+    pool: web::Data<PgPool>,
+) -> Result<impl Responder, Error> {
+    let uuid = parse_uuid(id.into_inner())?;
+    let mut conn = pool.get()?;
+
+    let graph = conn
+        .get_score_graph_by_id(uuid)?
+        .ok_or(Error::NotFound(uuid))?;
+
+    Ok(HttpResponse::Ok()
+        .content_type("text/turtle")
+        .message_body(graph))
+}
+
+#[get("/api/v1/score/{id}")]
+async fn get_score_json(
+    id: web::Path<String>,
+    pool: web::Data<PgPool>,
+) -> Result<impl Responder, Error> {
+    let uuid = parse_uuid(id.into_inner())?;
+    let mut conn = pool.get()?;
+
+    let score = conn
+        .get_score_json_by_id(uuid)?
+        .ok_or(Error::NotFound(uuid))?;
+
+    Ok(HttpResponse::Ok()
+        .content_type(mime::APPLICATION_JSON)
+        .message_body(score))
 }
 
-#[get("/score/{id}")]
-#[openapi(tags("score"))]
-#[openapi(id = "score")]
-#[openapi(summary = "Get dataset score")]
-async fn score(#[data] pool: Pool, id: String) -> Result<Json<Scores>, Rejection> {
-    async fn score(pool: Pool, id: String) -> Result<Json<Scores>, Error> {
-        let uuid = Uuid::parse_str(id.as_ref()).map_err(|_| Error::InvalidID(id))?;
-
-        let client: Client = pool.get().await?;
-
-        if let Some(graph_string) = get_graph_by_id(&client, uuid).await? {
-            let scores = ScoreGraph::parse(graph_string)?.score()?;
-            Ok(Json::from(scores))
-        } else {
-            Err(Error::NotFound(uuid))
-        }
+#[post("/api/v1/save/{id}")]
```

Comment:
X-API-KEY trenger vi vel her?

Reply:
Jepp, tenkte å ta det som fremtidig forbedring. Bare det nødvendige for å vise at ting funker i denne PRen.

Reply:
Det er vel et krav før denne skal ut i prod. Ser det ikke som en forbedring.

Reply:
Ait, la til auth nå